modules/api/src/main/java/org/apache/fluo/api/data/Bytes.java

```diff
@@ -355,6 +354,36 @@ public static final Bytes of(String s, Charset c) {
   }
 
   /**
+   * Checks if this has the passed prefix
+   * 
+   * @param prefix
+   * @return true or false
+   */
+  public boolean startsWith(Bytes prefix) {
+    for (int i = 0; i < prefix.length; i++) {
```

Comment:
prefix (and suffix in next method) should be checked for null with `Objects.requiresNonNull(prefix)`